### PR TITLE
OEL-1458: Activate block Social share module

### PR DESCRIPTION
**Pull request by abel-santos-corral**
### Description

Activate module oe_webtools_social_share and place block Social share.

### Change log

- Added: config/optional/block.block.showcase_socialshare.yml
- Changed: oe_showcase.info.yml

### CODE REVIEW

File: config/optional/block.block.showcase_socialshare.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+weight: 0
+provider: null
+plugin: social_share
+settings:
+  id: social_share
+  label: 'Social Share'
+  label_display: '0'
+  provider: oe_webtools_social_share
+visibility:
+  request_path:
```
Comment (drishu) [inline config/optional/block.block.showcase_socialshare.yml]: please enable this for all content types

File: oe_showcase.info.yml [hunk 2]
```diff
@@ -46,6 +46,7 @@ install:
   - oe_showcase_contact_forms
   - oe_showcase_user_profile
   - oe_webtools
+  - oe_webtools_social_share
```
Comment (drishu) [inline oe_showcase.info.yml:49]: add a post_update hook to enable this module

File: tests/src/ExistingSite/SmokeTest.php [hunk 3]
```diff
@@ -14,7 +14,11 @@ class SmokeTest extends ShowcaseExistingSiteTestBase {
    */
   public function testWebsiteIsUp(): void {
     $this->drupalGet('<front>');
-    $this->assertSession()->statusCodeEquals(200);
+    $assert_session = $this->assertSession();
+    $assert_session->statusCodeEquals(200);
+    // Assert only one Block social share and that is inside the content region.
+    $assert_session->elementsCount('css', 'div#block-showcase-socialshare', 1);
+    $assert_session->elementExists('xpath', '//div[@class="container mt-5"]/*//div[@id="block-showcase-socialshare"]');
```
Comment (drishu) [inline tests/src/ExistingSite/SmokeTest.php]: this line makes the above one redundant, rather:
-grab the parent container, the main page content, use cores cssSelect() or whatever you like and grab the blocks wrapper
-assert the block title is present
-do this with a helper in ShowcaseExistingSiteTestBase
-check for this in all the content type tests: page, news, event
-remove this code from the smoke test

**Review by drishu — CHANGES_REQUESTED** [review]

File: tests/src/ExistingSite/SmokeTest.php [hunk 4]
```diff
@@ -14,7 +14,8 @@ class SmokeTest extends ShowcaseExistingSiteTestBase {
    */
   public function testWebsiteIsUp(): void {
     $this->drupalGet('<front>');
-    $this->assertSession()->statusCodeEquals(200);
+    $assert_session = $this->assertSession();
```
Comment (drishu) [inline tests/src/ExistingSite/SmokeTest.php]: revert this change, no need to create a variable just for this

File: tests/src/ExistingSite/ShowcaseExistingSiteCreateEventTest.php [hunk 5]
```diff
@@ -113,6 +113,9 @@ public function testCreateEvent() {
     $assert_session->responseContains('image-test.png');
     $assert_session->responseContains('Starter Image test');
     $assert_session->responseContains('Starter Image test alt');
+
+    // Assert Social Share block.
+    $assert_session->elementExists('xpath', '//div[@class="container mt-5"]/*//div[contains(., "Share this page") and @id="block-showcase-socialshare"]');
```
Comment (drishu) [inline tests/src/ExistingSite/ShowcaseExistingSiteCreateEventTest.php]: checking for the id doesn't help, either check for the json, or just the title.

And in the end, now you have to change it in 3 places, so better with the helper method

**Review by drishu — CHANGES_REQUESTED** [review]

**Review by drishu — APPROVED** [review]

**Review by drishu — DISMISSED** [review]

**Review by brummbar — CHANGES_REQUESTED** [review]
Code is good but we need to override the template and use a `h3.fw-bold` as per Figma:
image

This is what we have:
image

**Comment by brummbar** [comment]
Just the title, the widget cannot and shouldn't be themed.

File: modules/oe_showcase_page/templates/oe-webtools-social-share.html.twig [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{#
+/**
+ * @file
+ * Override template for webtools social share.
+ *
+ * @see ./modules/contrib/oe_webtools/modules/oe_webtools_social_share/templates/oe_webtools_social_share.html.twig
```
Comment (brummbar) [inline modules/oe_showcase_page/templates/oe-webtools-social-share.html.twig]: Nit:
```suggestion
 * @see ./modules/contrib/oe_webtools/modules/oe_webtools_social_share/templates/oe-webtools-social-share.html.twig
```

**Review by brummbar — CHANGES_REQUESTED** [review]
A very very very small nit.
For the rest the code is good, but it points out that we really need a child theme in our profile.
I'll open the ticket, but this code can move forward now.

**Review by brummbar — APPROVED** [review]

**Review by drishu — APPROVED** [review]

**Pull request merged**